packages/api-v2/src/hooks/useTradingPlatformStatus.ts

```diff
@@ -0,0 +1,43 @@
+import { CFD_PLATFORMS, TRADING_PLATFORM_STATUS } from '@deriv/shared';
+import useAuthorizedQuery from '../useAuthorizedQuery';
+import useQuery from '../useQuery';
+
+type TPlatformStatus = Exclude<
+    NonNullable<ReturnType<typeof useQuery<'trading_platform_status'>>['data']>['trading_platform_status'][0],
+    undefined
+>;
+
+/** A custom hook that gets the list of statuses of ctrader dxtrade mt5 platform. */
+const useTradingPlatformStatus = () => {
+    const { data, ...rest } = useAuthorizedQuery(
+        'trading_platform_status',
+        {},
+        {
+            refetchInterval: 120000,
+        }
+    );
+
+    const tradingPlatformStatusData = data?.trading_platform_status;
+
+    /**
+     * Retrieves the status of a specified trading platform.
+     * @param platform The platform identifier (e.g., 'ctrader', 'dxtrade', 'mt5').
+     * @returns The status of the identified platform ('active', 'maintenance', 'unavailable').
+     */
+    const getPlatformStatus = (platform: string) => {
```

Comment:
```suggestion
    const getPlatformStatus = (platform: typeof CFD_PLATFORMS[keyof typeof CFD_PLATFORMS]) => {
```
Based on the comment i am assuming only values available in `CFD_PLATFORMS` is in the response of `trading_platform_status`. Using `string` primitive is too loose of a typing

Reply:
this is because only CFD platforms have statuses

when this hook is being called in cashier components, non-cfd platforms are passed (doughflow/p2p...) which will throw error when we type it as `typeof keyof typeof CFD_PLATFORMS` because `trading_platform_status` is only accepting mt5/ctrader/dxtrade

The checks `platform === CFD_PLATFORMS.MT5 `  are in the hook instead of the component for that reason